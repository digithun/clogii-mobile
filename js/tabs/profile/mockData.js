export const user = {
  id: 1,
  name: 'test',
  sex: 'M',
  facebookLinked: false
};
export const following = [{name: 'Art Nattapat'}, {name: 'Art Art Art'}, {name: 'Art Art Art'}, {name: 'Art Art Art'}, {name: 'Art Art Art'}, {name: 'Art Art Art'}];
export const follower = [{name: 'Art Nattapat', following: true}, {name: 'Art Art Art', following: false}, {name: 'Art Art Art', following: false}, {name: 'Art Art Art', following: true}, {name: 'Art Art Art', following: true}, {name: 'Art Art Art', following: false}];
export const myFan = [{name: 'Art Nattapat', candys: 1250}, {name: 'Art Art Art', candys: 1250}, {name: 'Art Art Art', candys: 1000}, {name: 'Art Art Art', candys: 789}, {name: 'Art Art Art', candys: 7013}, {name: 'Art Art Art', candys: 0}];
export const candys = 1890;
export const activity = [{
  activity: 'like',
  title: `แมค เดมอน ตะลุยอวกาศ`,
  outline: `The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร`,
  uri: 'http://3.bp.blogspot.com/-zW6wqY_1Me0/VYvJMOV4mcI/AAAAAAAAD-4/mB_AxhFoJH4/s1600/178491main_sig07-009-516.jpg',
  date: new Date(Date.now() - 5000)
},{
  activity: 'like',
  title: `แมค เดมอน ตะลุยอวกาศ`,
  outline: `The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร`,
  uri: 'http://3.bp.blogspot.com/-zW6wqY_1Me0/VYvJMOV4mcI/AAAAAAAAD-4/mB_AxhFoJH4/s1600/178491main_sig07-009-516.jpg',
  date: new Date(2016, 10, 8)
}, {
  activity: 'read',
  title: `แมค เดมอน ตะลุยอวกาศ`,
  outline: `The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร`,
  uri: 'http://www.fujisan.ne.jp/fit/th/images/ohishikouen430.jpg',
  date: new Date(2015, 10, 8)
}];

export const myClogs = [
  {
    title: 'Richy Rich! รวยมากนะ! รู้ยังคะทุกคน',
    authors: 'David Beckham',
    cover: require('./img/A.png'),
    categoryCover: require('./img/category/M.png'),
    views: 12300,
    likes: 1500,
    date: new Date(2015, 5, 24)
  },
  {
    title: 'Money Honey คุณชายหน้าตายกับยัยขี้งก',
    authors: 'สุดสาครนอย ไทรโยค',
    cover: require('./img/B.png'),
    categoryCover: require('./img/category/N.png'),
    views: 100,
    likes: 55,
    date: new Date(2016, 10, 8)
  }
];

export const bookmark = [
  {
    title: 'Richy Rich! รวยมากนะ! รู้ยังคะทุกคน',
    cover: require('./img/A.png'),
    categoryCover: require('./img/category/N.png'),
    bookmarkCount: 7
  },
  {
    title: 'Money Honey คุณชายหน้าตายกับยัยขี้งก',
    cover: require('./img/B.png'),
    categoryCover: require('./img/category/M.png'),
    bookmarkCount: 3
  }
];
