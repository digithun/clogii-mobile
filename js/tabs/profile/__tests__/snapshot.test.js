import React from 'react';
import renderer from 'react-test-renderer';
import {bookmark as mockBookmark, myClogs as mockMyClogs, user as mockUser} from '../mockData';
import * as mockData from '../mockData';

describe('BookmarkScreen', () => {
  const BookmarkScreen = require('../components/BookmarkScreen').default;
  it('render BookmarkScreen', () => {
    const tree = renderer.create(<BookmarkScreen bookmark={mockBookmark}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('CircleImageWithCategory', () => {
  const CircleImageWithCategory = require('../components/CircleImageWithCategory').default;
  it('render CircleImageWithCategory', () => {
    const tree = renderer.create(<CircleImageWithCategory
      source={{uri: 'x.jpg'}}
      categorySource={{uri: 'x.jpg'}}
      size={100}
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('JellyShopScreen', () => {
  const JellyShopScreen = require('../components/JellyShopScreen').default;
  it('render JellyShopScreen', () => {
    const tree = renderer.create(<JellyShopScreen/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('ModalSpinner', () => {
  const ModalSpinner = require('../components/ModalSpinner').default;
  it('render visible', () => {
    const tree = renderer.create(<ModalSpinner visible={true}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render invisible', () => {
    const tree = renderer.create(<ModalSpinner visible={false}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('MyClogScreen', () => {
  const MyClogScreen = require('../components/MyClogScreen').default;
  it('render MyClogScreen', () => {
    const tree = renderer.create(<MyClogScreen myClogs={mockMyClogs}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('ProfileCover', () => {
  const ProfileCover = require('../components/ProfileCover').default;
  const Text = require('react-native').Text;
  it('render default', () => {
    const tree = renderer.create(<ProfileCover />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render customSource', () => {
    const tree = renderer.create(<ProfileCover customSource={{uri: 'custom'}}/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render user cover with children', () => {
    const tree = renderer.create(<ProfileCover user={mockUser}><Text>User Cover</Text></ProfileCover>)
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('ProfileHeader', () => {
  const ProfileHeader = require('../components/ProfileHeader').default;
  it('render', () => {
    const tree = renderer.create(<ProfileHeader user={mockUser}/>)
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('UserContainer', () => {
  const UserContainer = require('../components/UserContainer');

  describe('FollowingScreen', () => {
      const FollowingScreen = UserContainer.FollowingScreen;
      it('render', () => {
        const tree = renderer.create(<FollowingScreen userList={mockData.following}/>);
        expect(tree.toJSON()).toMatchSnapshot();
      });
  });

  describe('MyFanScreen', () => {
      const MyFanScreen = UserContainer.MyFanScreen;
      it('render', () => {
        const tree = renderer.create(<MyFanScreen userList={mockData.myFan}/>);
        expect(tree.toJSON()).toMatchSnapshot();
      });
  });

  describe('FollowerScreen', () => {
      const FollowerScreen = UserContainer.FollowerScreen;
      it('render', () => {
        const tree = renderer.create(<FollowerScreen userList={mockData.follower}/>);
        expect(tree.toJSON()).toMatchSnapshot();
      });
  });
});

describe('ProfileScreen', () => {
  require('parse/react-native');
  it('');
  // const ProfileScreenComponent = require('../containers/ProfileScreen').Component;
});
