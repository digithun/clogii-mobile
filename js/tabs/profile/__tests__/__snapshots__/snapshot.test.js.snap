exports[`BookmarkScreen render BookmarkScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(250, 250, 250)",
      "flex": 1,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 1,
        "borderColor": "rgb(240, 240, 240)",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingTop": 5,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "opacity": 1,
          "paddingLeft": 20,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
          }
        } />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(209, 87, 92)",
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }>
        Bookmark
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "opacity": 1,
          "paddingRight": 20,
        }
      }
      testID={undefined} />
  </View>
  <ScrollView
    contentInset={
      Object {
        "bottom": 1354,
        "top": 0,
      }
    }
    data={
      Array [
        Object {
          "bookmarkCount": 7,
          "categoryCover": 1,
          "cover": 1,
          "index": 0,
          "title": "Richy Rich! รวยมากนะ! รู้ยังคะทุกคน",
          "willDelete": false,
        },
        Object {
          "bookmarkCount": 3,
          "categoryCover": 1,
          "cover": 1,
          "index": 1,
          "title": "Money Honey คุณชายหน้าตายกับยัยขี้งก",
          "willDelete": false,
        },
      ]
    }
    dataSource={
      ListViewDataSource {
        "items": 2,
      }
    }
    initialListSize={10}
    minContentHeight={1354}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={1000}
    onKeyboardDidHide={undefined}
    onKeyboardDidShow={undefined}
    onKeyboardWillHide={undefined}
    onKeyboardWillShow={undefined}
    onLayout={[Function]}
    onScroll={[Function]}
    pageSize={1}
    removeClippedSubviews={true}
    renderFooter={[Function]}
    renderRow={[Function]}
    renderSeparator={[Function]}
    scrollEventThrottle={50}
    scrollRenderAheadDistance={1000}
    stickyHeaderIndices={Array []}>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "padding": 13,
        }
      }
      testID={undefined}>
      <View
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        }>
        <Image
          categorySource={1}
          size={100}
          source={1}
          style={
            Array [
              Object {
                "borderRadius": 50,
                "height": 100,
                "width": 100,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "transparent",
                },
              ],
            ]
          } />
        <Image
          source={1}
          style={
            Object {
              "bottom": 3.125,
              "height": 25,
              "position": "absolute",
              "right": 3.125,
              "width": 25,
            }
          } />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 10,
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 5,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }>
            Richy Rich! รวยมากนะ! รู้ยังคะทุกคน
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingTop": 10,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }>
            7
             Bookmarks
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "resizeMode": "contain",
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "height": 1,
        }
      } />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "padding": 13,
        }
      }
      testID={undefined}>
      <View
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        }>
        <Image
          categorySource={1}
          size={100}
          source={1}
          style={
            Array [
              Object {
                "borderRadius": 50,
                "height": 100,
                "width": 100,
              },
              Array [
                undefined,
                Object {
                  "backgroundColor": "transparent",
                },
              ],
            ]
          } />
        <Image
          source={1}
          style={
            Object {
              "bottom": 3.125,
              "height": 25,
              "position": "absolute",
              "right": 3.125,
              "width": 25,
            }
          } />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "paddingLeft": 10,
          }
        }>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 5,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }>
            Money Honey คุณชายหน้าตายกับยัยขี้งก
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingTop": 10,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }>
            3
             Bookmarks
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "flex-end",
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "resizeMode": "contain",
            }
          } />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "height": 1,
        }
      } />
  </ScrollView>
</View>
`;

exports[`CircleImageWithCategory render CircleImageWithCategory 1`] = `
<View
  style={
    Object {
      "height": 100,
      "width": 100,
    }
  }>
  <Image
    categorySource={
      Object {
        "uri": "x.jpg",
      }
    }
    size={100}
    source={
      Object {
        "uri": "x.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 50,
          "height": 100,
          "width": 100,
        },
        Array [
          undefined,
          Object {
            "backgroundColor": "transparent",
          },
        ],
      ]
    } />
  <Image
    source={
      Object {
        "uri": "x.jpg",
      }
    }
    style={
      Object {
        "bottom": 3.125,
        "height": 25,
        "position": "absolute",
        "right": 3.125,
        "width": 25,
      }
    } />
</View>
`;
